src/bitset.hpp

```diff
@@ -76,14 +91,32 @@ class Bitset {
   std::optional<uint32_t> SingletonOption() const;
   // Get the number of 1s.
   size_t Count() const;
+  // Output as string of comma-separated indices.
   std::string ToIndexSetString() const;
 
+  // ** Subsplit Methods
   // These methods require the bitset to be a "subsplit bitset" of even length,
   // consisting of two equal sized "chunks" representing the two sides of the
   // subsplit.
   //
+  // Subsplit Constructor: each argument represents one of the children of the subsplit.
+  // Each child templated after Bitset constructor.
+  // Build a Subsplit bitset out of a compatible pair of chunks.
+  static Bitset Subsplit(const Bitset &child_0, const Bitset &child_1);
+  // Builds Children from string of "1" and "0" characters.
+  static Bitset Subsplit(const std::string child_0, const std::string child_1);
+  // Builds Children of size `n` with only indices in the child vectors set to true.
+  static Bitset Subsplit(const SizeVector child_0, const SizeVector child_1,
+                         const size_t n);
+  // Comparator for Subsplit.  This method produces a topological sort.
```

Comment:
Re "This method produces a topological sort." this is awesome but either needs more context and a proof, or removal. (It'll go somewhere!)

Reply:
I'm quite happy to have the comparator! I just meant the statement about topological sort.

It does look like it needs a test though?